core/java/iesi-core/src/main/java/io/metadew/iesi/metadata/configuration/script/ScriptVersionConfiguration.java

```diff
@@ -173,26 +225,40 @@ private String updateStatement(ScriptVersion scriptVersion){
+    public boolean exists(ScriptVersionKey scriptVersionKey) {
+        String query = "select SCRIPT_ID, SCRIPT_VRS_NB, SCRIPT_VRS_DSC from " + getMetadataRepository().getTableNameByLabel("ScriptVersions")
+                + " where SCRIPT_ID = " + SQLTools.getStringForSQL(scriptVersionKey.getScriptKey().getScriptId()) +
+                " and SCRIPT_VRS_NB = " + SQLTools.getStringForSQL(scriptVersionKey.getScriptKey().getScriptVersion()) +
+                " and DELETED_AT = 'NA' ;";
+        CachedRowSet cachedRowSet = getMetadataRepository().executeQuery(query, "reader");
+        return cachedRowSet.size() >= 1;
     }
 
-    public boolean exists(ScriptVersionKey scriptVersionKey) {
+    public boolean existsDeleted(ScriptVersionKey scriptVersionKey) {
         String query = "select SCRIPT_ID, SCRIPT_VRS_NB, SCRIPT_VRS_DSC from " + getMetadataRepository().getTableNameByLabel("ScriptVersions")
                 + " where SCRIPT_ID = " + SQLTools.getStringForSQL(scriptVersionKey.getScriptKey().getScriptId()) +
-                " and SCRIPT_VRS_NB = " + SQLTools.getStringForSQL(scriptVersionKey.getScriptKey().getScriptVersion()) + ";";
+                " and SCRIPT_VRS_NB = " + SQLTools.getStringForSQL(scriptVersionKey.getScriptKey().getScriptVersion()) +
+                " and DELETED_AT = " + SQLTools.getStringForSQL(scriptVersionKey.getScriptKey().getDeletedAt()) + " ;";
         CachedRowSet cachedRowSet = getMetadataRepository().executeQuery(query, "reader");
         return cachedRowSet.size() >= 1;
     }
 
     public String getInsertStatement(ScriptVersion scriptVersion) {
         return "INSERT INTO " + getMetadataRepository().getTableNameByLabel("ScriptVersions") +
-                " (SCRIPT_ID, SCRIPT_VRS_NB, SCRIPT_VRS_DSC) VALUES (" +
+                " (SCRIPT_ID, SCRIPT_VRS_NB, SCRIPT_VRS_DSC, DELETED_AT) VALUES (" +
                 SQLTools.getStringForSQL(scriptVersion.getMetadataKey().getScriptKey().getScriptId()) + ", " +
                 SQLTools.getStringForSQL(scriptVersion.getMetadataKey().getScriptKey().getScriptVersion()) + ", " +
-                SQLTools.getStringForSQL(scriptVersion.getDescription()) + ");";
+                SQLTools.getStringForSQL(scriptVersion.getDescription()) + ", " +
+                " 'NA' );";
+    }
+
+    public String getUpdateStatement(ScriptVersion scriptVersion) {
+        return " UPDATE " + getMetadataRepository().getTableNameByLabel("ScriptVersions") + " SET " +
+                " SCRIPT_VRS_DSC = " + SQLTools.getStringForSQL(scriptVersion.getDescription()) + ", " +
+                " LAST_MODIFIED_BY = " + SQLTools.getStringForSQL(scriptVersion.getLastModifiedBy()) + ", " +
+                " LAST_MODIFIED_AT = " + SQLTools.getStringForSQL(scriptVersion.getLastModifiedAt()) +
+                " WHERE SCRIPT_ID = " + SQLTools.getStringForSQL(scriptVersion.getMetadataKey().getScriptKey().getScriptId()) +
+                " AND DELETED_AT = 'NA' ;";
```

Comment:
Doesn't ScriptVersionKey contain deletedAt? don't e need to use this value here?